CONTRIBUTING.md

```diff
@@ -83,7 +83,7 @@ Instead of relying directly on `black` however, we rely on
 [ufmt](https://github.com/omnilib/ufmt), for compatibility reasons with Facebook
 internal infrastructure.
 
-To format your code, install `ufmt` with `pip install ufmt==1.3.2 black==22.3.0 usort==1.0.2` and use e.g.:
+To format your code, install `ufmt` with `pip install ufmt==1.3.2 black==22.3.0 usort==1.0.2 flake8=6.0.0` and use e.g.:
```

Comment:
Since this section is about formatting, can we not include `flake8` here, but rather have a separate installation command below where `flake8` usage is explained? Alternatively, we could also rephrase this to not start with 

> To format your code, install `ufmt`

Reply:
I split up the formatting and linting sections. 

Also promoted the "typing" section to same level as formatting/linting and unit tests.  Mypy operates independently from fromatting/linting and I wanted to avoid confusion in the pre-commit hooks section.

lmk if these changes look good. 

CI is failing same as other PRs. Issues seem unrelated, but I'm fine waiting until everything is green again to avoid compounding any CI issues.

Reply:
> CI is failing same as other PRs. Issues seem unrelated, but I'm fine waiting until everything is green again to avoid compounding any CI issues.

Just as a heads up, this will take some time. See #7299.

Reply:
The previous issue that led to these type hints being removed is partially hidden behind internal meta tools, but seems to basically be https://github.com/pytorch/vision/pull/4631/files#r744161323 . 

Imo it would be odd to sacrifice typing in torchvision just to avoid adding some type ignore or assert in the 3rd party code. However, lmk if that's not the case.